src/OpenCV/Features2d.hsc

```diff
@@ -285,6 +294,213 @@ orbDetectAndCompute orb img mbMask = unsafeWrapException $ do
 
           pure $ Right (V.fromList keypoints, relaxMat descriptors)
 
+--------------------------------------------------------------------------------
+-- BLOB - Binary Large OBject
+--------------------------------------------------------------------------------
+
+-- Internally, a SimpleBlobDetector is a pointer to a @cv::Ptr<cv::SimpleBlobDetector>@, which in turn points
+-- to an actual @cv::SimpleBlobDetector@ object.
+newtype SimpleBlobDetector = SimpleBlobDetector {unSimpleBlobDetector :: ForeignPtr C'Ptr_SimpleBlobDetector}
+
+type instance C SimpleBlobDetector = C'Ptr_SimpleBlobDetector
+
+instance WithPtr SimpleBlobDetector where
+    withPtr = withForeignPtr . unSimpleBlobDetector
+
+instance FromPtr SimpleBlobDetector where
+    fromPtr = objFromPtr SimpleBlobDetector $ \ptr ->
+                [CU.block| void {
+                  cv::Ptr<cv::SimpleBlobDetector> * simpleBlobDetector_ptr_ptr = $(Ptr_SimpleBlobDetector * ptr);
+                  simpleBlobDetector_ptr_ptr->release();
+                  delete simpleBlobDetector_ptr_ptr;
+                }|]
+
+data SimpleBlobDetectorParams
+   = SimpleBlobDetectorParams
+   { blob_blobColor :: !Word8
+   , blob_filterByArea :: !Bool
+     -- ^ Extracted blobs have an area between 'minArea' (inclusive) and
+     --   'maxArea' (exclusive).
+   , blob_filterByCircularity :: !Bool
+     -- ^ Extracted blobs have circularity
+     --   @(4 * pi * Area)/(perimeter * perimeter)@ between 'minCircularity'
+     --   (inclusive) and 'maxCircularity' (exclusive).
+   , blob_filterByColor :: !Bool
+     -- ^ This filter compares the intensity of a binary image at the center of
+     --   a blob to 'blobColor'. If they differ, the blob is filtered out. Use
+     --   @blobColor = 0@ to extract dark blobs and @blobColor = 255@ to extract
+     --   light blobs.
+   , blob_filterByConvexity :: !Bool
+     -- ^ Extracted blobs have convexity (area / area of blob convex hull) between
+     --   'minConvexity' (inclusive) and 'maxConvexity' (exclusive).
+   , blob_filterByInertia :: !Bool
+     -- ^ Extracted blobs have this ratio between 'minInertiaRatio' (inclusive)
+     --   and 'maxInertiaRatio' (exclusive).
+   , blob_maxArea :: !Float
+   , blob_maxCircularity :: !Float
+   , blob_maxConvexity :: !Float
+   , blob_maxInertiaRatio :: !Float
+   , blob_maxThreshold :: !Float
+   , blob_minArea :: !Float
+   , blob_minCircularity :: !Float
+   , blob_minConvexity :: !Float
+   , blob_minDistBetweenBlobs :: !Float
+   , blob_minInertiaRatio :: !Float
+   , blob_minRepeatability :: !Int32
+   , blob_minThreshold :: !Float
+   , blob_thresholdStep :: !Float
+   }
+
+defaultSimpleBlobDetectorParams :: SimpleBlobDetectorParams
+defaultSimpleBlobDetectorParams =
+    SimpleBlobDetectorParams
+      { blob_blobColor = 0
```

Comment:
Here I would also align the curly braces `{` with the constructor name. So in this case 2 places to the left.